appeals/web/src/server/appeals/appeal-details/appellant-case/__tests__/appellant-case.test.js

```diff
@@ -1162,6 +1163,212 @@ describe('appellant-case', () => {
 		});
 	});
 
+	describe('GET /appellant-case/valid/date', () => {
+		it('should render the 500 error page if required data is not present in the session', async () => {
+			const response = await request.get(
+				`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`
+			);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should render the valid due date page if required data is present in the session', async () => {
+			await request.post(`${baseUrl}/1${appellantCasePagePath}`).send({
+				reviewOutcome: 'valid'
+			});
+
+			const response = await request.get(
+				`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`
+			);
+			const element = parseHtml(response.text);
+
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+	});
+
+	describe('POST /appellant-case/valid/date', () => {
+		beforeEach(async () => {
+			await request.post(`${baseUrl}/1${appellantCasePagePath}`).send({
+				reviewOutcome: 'valid'
+			});
+		});
+
+		afterEach(() => {
+			nock.cleanAll();
+		});
+
+		it('should re-render the valid date page with the expected error message if no date was provided', async () => {
+			const response = await request
+				.post(`${baseUrl}/1${appellantCasePagePath}${validOutcomePagePath}${validDatePagePath}`)
+				.send({
+					'due-date-day': '',
+					'due-date-month': '',
+					'due-date-year': ''
+				});
+
+			expect(response.statusCode).toBe(200);
+			const element = parseHtml(response.text);
+			expect(element.innerHTML).toMatchSnapshot();
+		});
+
+		it('should re-render the valid date page with the expected error message if provided date is not in the future', async () => {
```

Comment:
Would it be worth breaking the test into parts specific to what it is testing? This would avoid overlap.